test/e2e/annotations/modsecurity.go

```diff
@@ -216,4 +216,78 @@ var _ = framework.DescribeAnnotation("modsecurity owasp", func() {
 			Expect().
 			Status(http.StatusForbidden)
 	})
+
+	ginkgo.It("should enable modsecurity when enable-owasp-modsecurity-crs is set to true", func() {
+		host := "modsecurity.foo.com"
+		nameSpace := f.Namespace
+
+		snippet := `SecRuleEngine On
+		SecRequestBodyAccess On
+		SecAuditEngine RelevantOnly
+		SecAuditLogParts ABIJDEFHZ
+		SecAuditLog /dev/stdout
+		SecAuditLogType Serial
+		SecRule REQUEST_HEADERS:User-Agent \"block-ua\" \"log,deny,id:107,status:403,msg:\'UA blocked\'\"`
+
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/modsecurity-snippet": snippet,
+		}
+
+		ing := framework.NewSingleIngress(host, "/", host, nameSpace, framework.EchoService, 80, annotations)
+		f.EnsureIngress(ing)
+
+		f.UpdateNginxConfigMapData("enable-modsecurity", "true")
+		f.UpdateNginxConfigMapData("enable-owasp-modsecurity-crs", "true")
+
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return strings.Contains(server, "SecRuleEngine On")
+			})
+
+		f.HTTPTestClient().
+			GET("/").
+			WithHeader("Host", host).
+			WithHeader("User-Agent", "block-ua").
+			Expect().
+			Status(http.StatusForbidden)
+	})
+
+	ginkgo.It("should enable modsecurity through the config map", func() {
+		host := "modsecurity.foo.com"
+		nameSpace := f.Namespace
+
+		snippet := `SecRequestBodyAccess On
+		SecAuditEngine RelevantOnly
+		SecAuditLogParts ABIJDEFHZ
+		SecAuditLog /dev/stdout
+		SecAuditLogType Serial
+		SecRule REQUEST_HEADERS:User-Agent \"block-ua\" \"log,deny,id:107,status:403,msg:\'UA blocked\'\"`
+
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/modsecurity-snippet": snippet,
+		}
+
+		ing := framework.NewSingleIngress(host, "/", host, nameSpace, framework.EchoService, 80, annotations)
+		f.EnsureIngress(ing)
+
+		f.UpdateNginxConfigMapData("enable-modsecurity", "true")
+		f.UpdateNginxConfigMapData("enable-owasp-modsecurity-crs", "true")
+		f.UpdateNginxConfigMapData("SecRuleEngine", "On")
```

Comment:
This is not a valid configuration for the configmap. What are you trying to do?

Reply:
I'm trying to move `SecRuleEngine On` from the snippet which would go into the Ingress resource into the Ingress controller so that it can be set globally as opposed to in multiple ingress resources

Reply:
@gabatwork not sure that makes sense. It would be easier to just mount a custom `/etc/nginx/modsecurity/modsecurity.conf` file. That requires no changes to the ingress controller.

Reply:
this is what I am referring to: https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#modsecurity-snippet

but it doesn't seem to work, I noticed that the format is incorrect, but I'm not sure how I could pass into the config map here, in my working ingress resource it looks something like this: 
`    nginx.ingress.kubernetes.io/modsecurity-snippet: |
        SecRuleEngine On
        SecRuleRemoveById 920170`
so in theory, I should be able to have the following in my controller config right?
`modsecurity-snippet: |
        SecRuleEngine On
        SecRuleRemoveById 920170`

Reply:
> It would be easier to just mount a custom `/etc/nginx/modsecurity/modsecurity.conf` file

Any chance that solution is documented then? At the moment it modsecurity does not look too usable due to https://github.com/kubernetes/ingress-nginx/issues/6307 it would be awesome if there was a solution simpler than a single  configuration directive.